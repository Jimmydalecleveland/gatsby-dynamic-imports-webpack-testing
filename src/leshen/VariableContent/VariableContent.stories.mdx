import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Typography from '../Typography'
import Button from '../Button'
import Columns from '../Columns'
import Column from '../Columns/Column'
import Link from '../Link'
import VariableContent, { ticketID } from '.'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|VariableContent' component={VariableContent} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# VariableContent
<Preview>
  <Story name="default">
    <div>
      <VariableContent
        heading="A super heading"
        subheading="A great subheading"
        centeredContent
        mainContent={
          <>
            <Typography>
              CenturyLink Price for Life Internet Plans ensure your monthly
              internet rate stays the same when you remain on the same internet
              plan at the same address. No contract, no rate hikes.
            </Typography>
            <Typography>
              See what CenturyLink Services are in your area:
            </Typography>
            <Typography>
             See what CenturyLink Services are in your area: <Link to="/section-link"> Call Now</Link>
            </Typography>
          </>
        }
      />
      <VariableContent
        mainContent={
          <>
            <Typography variant="h2">A very basic Section</Typography>
            <Typography>
              Natoque a etiam egestas parturient enim cum semper litora lacinia
              habitant a a praesent. Natoque a etiam egestas parturient enim cum
            </Typography>
          </>
        }
      />
    </div>
  </Story>
</Preview>

## Props
<Props of={VariableContent} />

## with Columns
<Preview>
  <Story name="with Columns">
    <VariableContent
      heading="A super heading"
      subheading="A great subheading"
      mainContent={
        <>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Typography>
            See what CenturyLink Services are in your area:
          </Typography>
        </>
      }
    >
      <Columns>
        <Column>
          <Typography variant="h3">Column Heading</Typography>
          <Typography>
            Internet rate stays the same when you remain on the same internet
            plan at the same address. No contract, no rate hikes.
          </Typography>
          <Button>Get your good stuff here!</Button>
        </Column>
        <Column>
          <Typography variant="h3">Column Heading</Typography>
          <Typography>
            Hodor hodor! Hodor? Hodor, hodor. Hodor. Hodor. Hodor, hodor. Hodor.
            Hodor, hodor. Hodor. HODOR HODOR! HODOR hodor, hodor. Hodor hodor!
            Hodor? Hodor! Hodor hodor. Hodor hodor HODOR! Hodor. Hodor, hodor,
            hodor , hodor. Hodor hodor HODOR! Hodor.
          </Typography>
          <Button>Hodor!</Button>
        </Column>
        <Column>
          <Typography variant="h3">Column Heading</Typography>
          <Typography>
            CenturyLink Price for Life Internet Plans ensure your monthly
            internet rate stays the same when you remain on the same internet.
          </Typography>
          <Button>Get your good stuff here!</Button>
        </Column>
      </Columns>
    </VariableContent>
  </Story>
</Preview>

## with Prominent
<Preview>
  <Story name="Prominent">
    <div>
      <VariableContent
        heading="A super heading"
        subheading="A great subheading"
        backgroundColor="Prominent"
        centeredContent
        mainContent={
          <>
            <Typography>
              CenturyLink Price for Life Internet Plans ensure your monthly
              internet rate stays the same when you remain on the same internet
              plan at the same address. No contract, no rate hikes.
            </Typography>
            <Typography>
              See what CenturyLink Services are in your area:
            </Typography>
            <Typography>
             See what CenturyLink Services are in your area: <Link to="/section-link"> Call Now</Link>
          </Typography>
          </>
        }
      />
    </div>
  </Story>
</Preview>