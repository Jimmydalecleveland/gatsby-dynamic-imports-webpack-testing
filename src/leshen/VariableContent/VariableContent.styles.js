import styled from "@emotion/styled"
import { css } from "@emotion/core"
import { stripUnit } from "polished"

import fluidRange from "../utils/fluidRange"

export const Intro = styled.header(
  ({ theme, centeredContent, hasChildren }) => css`
    text-align: ${centeredContent ? "center" : "left"};
    ${
      hasChildren &&
      fluidRange("margin-bottom", [
        stripUnit(theme.spacing.lg),
        stripUnit(theme.spacing.xxl),
      ])
    };

  ${
    !centeredContent &&
    `
      > .svg-image {
        margin-left: auto;
        margin-right: auto;
      }
    `
  }

    @media (min-width: ${theme.breakpoints.lg}) {
      width: ${centeredContent ? "60%" : "80%"};
      margin-left: auto;
      margin-right: auto;
    }
  `
)

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: ${({ centeredContent }) =>
    centeredContent ? "center" : "flex-start"};
`
