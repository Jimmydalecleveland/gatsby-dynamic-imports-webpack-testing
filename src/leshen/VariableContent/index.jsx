import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './VariableContent.styles'
import Section from '../Section'
import Img from '../Image'
import Typography from '../Typography'
import Breadcrumbs from '../Breadcrumbs'

const VariableContent = ({
  heading,
  subheading,
  headingLevel,
  mainContent,
  image,
  children,
  alignImageToBottom,
  centeredContent,
  breadcrumbProps,
  backgroundColor,
  isImageCritical,
  ...props
}) => (
  <Section backgroundColor={backgroundColor} {...props}>
    {breadcrumbProps && (
      <Breadcrumbs {...breadcrumbProps} backgroundColor={backgroundColor} />
    )}
    <Styled.Intro
      hasChildren={Boolean(children && heading)}
      centeredContent={centeredContent}
    >
      {image && !alignImageToBottom && (
        <Img data={image} isImageCritical={isImageCritical} />
      )}
      {heading && (
        <Typography variant={`h${headingLevel}`}>{heading}</Typography>
      )}
      {subheading && (
        <Typography
          variant={`h${headingLevel + 3}`}
          rendersAs={`h${headingLevel + 1}`}
        >
          {subheading}
        </Typography>
      )}
      <Styled.Content centeredContent={centeredContent}>
        {mainContent}
      </Styled.Content>
      {image && alignImageToBottom && (
        <Img data={image} isImageCritical={isImageCritical} />
      )}
    </Styled.Intro>
    {children}
  </Section>
)

VariableContent.defaultProps = {
  subheading: '',
  headingLevel: 2,
  mainContent: null,
  children: null,
  image: null,
  alignImageToBottom: false,
  centeredContent: false,
  backgroundColor: null,
  breadcrumbProps: null,
  isImageCritical: false,
}

VariableContent.propTypes = {
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  headingLevel: PropTypes.oneOf([1, 2]),
  image: PropTypes.shape({
    fluid: PropTypes.object,
    fixed: PropTypes.object,
    svg: PropTypes.object,
    context: PropTypes.object,
  }),
  alignImageToBottom: PropTypes.bool,
  centeredContent: PropTypes.bool,
  mainContent: PropTypes.node,
  children: PropTypes.node,
  backgroundColor: PropTypes.string,
  breadcrumbProps: PropTypes.shape({
    crumbs: PropTypes.string,
    path: PropTypes.string,
    siteURL: PropTypes.string,
  }),
  isImageCritical: PropTypes.bool,
}

export default VariableContent

export const ticketID = 'CMS-180'
