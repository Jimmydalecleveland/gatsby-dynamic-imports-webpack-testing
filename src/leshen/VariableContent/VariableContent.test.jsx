import React from 'react'
import { render } from 'setup-tests'
import VariableContent from '.'

describe('VariableContent', () => {
  test('VariableContent should render', () => {
    const { container } = render(
      <VariableContent
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
