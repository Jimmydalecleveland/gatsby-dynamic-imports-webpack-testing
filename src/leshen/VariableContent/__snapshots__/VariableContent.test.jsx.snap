// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableContent VariableContent should render 1`] = `
.emotion-8 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
}

@media (max-width:480px) {
  .emotion-8 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-8 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-8 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-8 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-8 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-8 {
    padding-bottom: 88px;
  }
}

.emotion-6 {
  text-align: left;
}

.emotion-6 > .svg-image {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width:940px) {
  .emotion-6 {
    width: 80%;
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-0 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-2 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<section
  class="emotion-8 emotion-9"
>
  <header
    class="emotion-6 emotion-7"
  >
    <h2
      class="emotion-0 emotion-1"
    >
      Heading
    </h2>
    <h3
      class="emotion-2 emotion-3"
    >
      Subheading
    </h3>
    <div
      class="emotion-4 emotion-5"
    >
      <p>
        Main Content
      </p>
    </div>
  </header>
</section>
`;
